src/nspawn/nspawn.c

```diff
@@ -5566,7 +5566,8 @@ static int run_container(
                         } else if (!isempty(arg_background))
                                 (void) pty_forward_set_background_color(forward, arg_background);
 
-                        set_window_title(forward);
+                        if (shall_set_terminal_title())
```

Comment:
I think it's better to move the check into set_window_title?

Reply:
I can move it there, but it seems to be inconsistent – current code calls shall_tint_background() in main, not in terminal_tint_color().